tests/workflows/basic_api_node_with_timeout/test_timeout.py

```diff
@@ -0,0 +1,70 @@
+from unittest.mock import Mock, patch
+
+from vellum.workflows.constants import APIRequestMethod
+from vellum.workflows.nodes.displayable.bases.api_node.node import BaseAPINode
+
+
+class TestAPINodeTimeout:
+    """Test timeout functionality in BaseAPINode."""
+
+    def _setup_vellum_mocks(self, node):
+        """Helper method to set up common vellum client mocks."""
+        mock_context = Mock()
+        mock_vellum_client = Mock()
+        mock_vellum_response = Mock()
+        mock_vellum_response.json_ = {"result": "success"}
+        mock_vellum_response.headers = {"content-type": "application/json"}
+        mock_vellum_response.status_code = 200
+        mock_vellum_response.text = '{"result": "success"}'
+        mock_vellum_client.execute_api.return_value = mock_vellum_response
+        mock_context.vellum_client = mock_vellum_client
+        node._context = mock_context
+        return mock_vellum_client, mock_vellum_response
+
+    @patch("vellum.workflows.nodes.displayable.bases.api_node.node.Session")
+    def test_local_execute_api_with_timeout(self, mock_session):
+        """Test that timeout is passed to local requests session."""
+        mock_session_instance = Mock()
+        mock_session.return_value.__enter__.return_value = mock_session_instance
+
+        mock_response = Mock()
+        mock_response.json.return_value = {"result": "success"}
+        mock_response.headers = {"content-type": "application/json"}
+        mock_response.status_code = 200
+        mock_response.text = '{"result": "success"}'
+        mock_session_instance.send.return_value = mock_response
+
+        node: BaseAPINode = BaseAPINode()
+        node.url = "https://example.com"
+
+        node._local_execute_api(
```

Comment:
[nit] Instead of calling private methods in tests, could we run `node.run()`? testing private methods makes it more inflexible for us to make changes on non-user facing logic